armi/reactor/reactors.py

```diff
@@ -1862,21 +1861,14 @@ def updateAxialMesh(self):
         # depending on what makes the most sense
         refAssem = self.refAssem
         refMesh = self.findAllAxialMeshPoints([refAssem])
-        avgHeight = average1DWithinTolerance(
-            numpy.array(
-                [
-                    [
-                        h
-                        for b in a
-                        for h in [(b.p.ztop - b.p.zbottom) / b.p.axMesh] * b.p.axMesh
-                    ]
-                    for a in self
-                    if self.findAllAxialMeshPoints([a]) == refMesh
-                ]
-            )
+        meshHeights = numpy.array(
+            [
+                h
+                for b in refAssem
+                for h in [(b.p.ztop - b.p.zbottom) / b.p.axMesh] * b.p.axMesh
+            ]
```

Comment:
Same story as above with respect to mesh averaging. I would like to test to see if htis change can be reverted before merging. If not, the viability of this approach could be questionable because changing the way `core.p.axialMesh` is computed could have lots of downstream effects.